stats/kafka/collector.go

```diff
@@ -155,19 +155,15 @@ func (c *Collector) pushMetrics() {
 
 	// Send the samples
 	c.logger.Debug("Kafka: Delivering...")
-
+	
+	var msgList []*sarama.ProducerMessage
 	for _, sample := range formattedSamples {
 		msg := &sarama.ProducerMessage{Topic: c.Config.Topic.String, Value: sarama.StringEncoder(sample)}
-		partition, offset, err := c.Producer.SendMessage(msg)
-		if err != nil {
-			c.logger.WithError(err).Error("Kafka: failed to send message.")
-		} else {
-			c.logger.WithFields(logrus.Fields{
-				"partition": partition,
-				"offset":    offset,
-			}).Debug("Kafka: message sent.")
-		}
+		msgList = append(msgList, msg)
 	}
+	// Send message in batch
+	c.Producer.SendMessages(msgList)
```

Comment:
> Is there any way to have logging while using `c.Producer.SendMessages(msgList)` on Line 165?

That is, what happens if this line somehow fails?

Reply:
Good point! Let me have a check. Thx @mostafa !

Reply:
Hi @RoySunnySean007 any movement on this change?